src/main/java/build/buildfarm/backplane/Backplane.java

```diff
@@ -281,4 +282,21 @@ boolean pollOperation(QueueEntry queueEntry, ExecutionStage.Value stage, long re
 
   /** Set expiry time for digests */
   void updateDigestsExpiry(Iterable<Digest> digests) throws IOException;
+
+  /**
+   * Updates the read count for CAS entries based on the provided stream of digest and count.
+   *
+   * @param casReadCountStream A Stream of Digest and its corresponding read count.
+   * @return A Map containing the updated read counts for the specified CAS entries.
+   */
+  Map<String, Integer> updateCasReadCount(Stream<Map.Entry<Digest, Integer>> casReadCountStream)
+      throws IOException;
+
+  /**
+   * Removes the CAS read count entries from the storage.
+   *
+   * @param digestsToBeRemoved CAS entries for which each read count needs to be removed.
+   * @return total count of cas read count entries removed.
+   */
+  int removeCasReadCountEntries(Stream<Digest> digestsToBeRemoved) throws IOException;
```

Comment:
Why are you using a Stream as an interface specification? Typically these are Iterables

Reply:
Changed them to Iterable, as it is more conventional .

Used Stream to leverage its functional programming capabilities and lazy evaluation, as I was performing transformations on each element. These transformations are relatively simple and don't benefit from the features offered by streams.